src/forms/FormControl.scss

```diff
@@ -57,26 +52,57 @@
 
   &[disabled] {
     color: var(--color-primer-fg-disabled);
+    cursor: not-allowed;
     background-color: var(--color-input-disabled-bg);
     border-color: var(--color-border-default);
-    -webkit-text-fill-color: var(--color-primer-fg-disabled);
     opacity: 1;
+    -webkit-text-fill-color: var(--color-primer-fg-disabled);
   }
 
   &:not(:focus)[invalid] {
     border-color: var(--color-danger-emphasis);
   }
 
+  &:not([type='checkbox']):not([type='radio']):focus {
+    @include focusBoxShadowInset;
+
+    // remove fallback :focus if :focus-visible is supported
+    &:not(:focus-visible) {
+      border-color: transparent;
+
+      @include focusBoxShadowInset(1px, transparent);
+    }
+  }
+
+  // default focus state
+  &:focus-visible {
+    @include focusBoxShadowInset;
+  }
+
+  &.FormControl--inset {
```

Comment:
Any thoughts on calling this inset instead of contrast? I might swap this back to contrast just for parity with PRC.

Reply:
I struggled with this one a bit! I need a class that can be shared across all form controls since so much logic is shared. Is there another term to describe formy type things that isn't too specific (like `Input`)? Maybe `FormField--etcetc` (thinking out loud here).

Reply:
I can't think of anything better than "FormField". It's tough to be descriptive without being too specific.

I'll comment again if I have any ideas.